puppet/modules/slave/manifests/init.pp

```diff
@@ -123,148 +68,25 @@
-
-    package { 'bower':
-      ensure   => '1.7.9',
-      provider => npm,
-    }
-    package { 'grunt-cli':
-      ensure   => present,
-      provider => npm,
-    }
-
-    # temporary dir
-    file { "${homedir}/tmp":
-      ensure => directory,
-      owner  => 'jenkins',
-      group  => 'jenkins',
-      mode   => '0775',
-    }
-
-    # Cleanup temporary dir
-    file { '/etc/cron.daily/npm_tmp_cleaner':
-      ensure  => file,
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0755',
-      content => template('slave/npm_cleaner.sh.erb'),
-    }
-  }
-
-  # Needed for integration tests with headless chrome and Selenium
-  if $facts['os']['family'] == 'RedHat' {
-    include epel
-
-    package { ['chromium', 'chromedriver']:
-      ensure  => latest,
-      require => Class['epel'],
-    }
-  }
-
-  # Needed for foreman-selinux testing
-  if $facts['os']['family'] == 'RedHat' {
-    ensure_packages(['selinux-policy-devel'])
-  }
-
-  # needed by katello gem dependency qpid_proton
-  # for katello-agent messaging
-  if $facts['os']['family'] == 'RedHat' {
-    package { 'qpid-proton-cpp-devel':
-      ensure => latest,
-    }
-  }
-
-  # Needed by foreman_openscap gem dependency OpenSCAP
-  if $facts['os']['family'] == 'RedHat' {
-    package { 'openscap':
-      ensure => latest,
-    }
-  }
-
-  # Increase OS limits, RH OSes ship them by default
-  if $facts['os']['family'] == 'RedHat' {
-    file { '/etc/security/limits.d':
-      ensure  => directory,
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0755',
-      recurse => true,
-      purge   => true,
-    }
-
-    file { '/etc/security/limits.d/90-nproc.conf':
-      ensure  => file,
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0644',
-      content => template('slave/90-nproc.conf.erb'),
-    }
-  }
-
-  # Java
-  include slave::java
-
-  # Databases
-  include slave::mysql
-  include slave::postgresql
-  # sqlite3 support was dropped in Foreman 2.1
-  slave::db_config { 'sqlite3':
-    ensure => absent,
-  }
-
-  # RVM
-  include slave::rvm
-
   # Packaging
-  case $facts['os']['family'] {
-    'RedHat': {
-      class { 'slave::packaging::rpm':
-        homedir          => $homedir,
-        koji_certificate => $koji_certificate,
-      }
-      contain slave::packaging::rpm
-    }
-    'Debian': {
-      class { 'slave::packaging::debian':
-        uploader  => $uploader,
-        user      => 'jenkins',
-        workspace => $workspace,
-      }
-      contain slave::packaging::debian
+  if $packaging {
+    class {'slave::packaging':
+      koji_certificate => $koji_certificate,
+      uploader         => $uploader,
+      homedir          => $homedir,
+      workspace        => $workspace,
     }
-    default: {}
   }
 
-  if $facts['os']['architecture'] in ['x86_64', 'amd64'] and !$is_el8 {
+  if $facts['os']['architecture'] in ['x86_64', 'amd64'] and !$facts['os']['release']['major'] == '8' {
```

Comment:
we don't support debian 8, and I moved the `is_el8` definition away out of this class anyways :woman_shrugging: